scripts/compute_csa.sh

```diff
@@ -0,0 +1,249 @@
+  sct_label_utils -i ${FILESEG}.nii.gz -disc ${FILELABEL}.nii.gz -o ${FILESEG}_labeled.nii.gz
+}
+
+
+# Copy GT spinal cord disc labels (located under derivatives/labels)
+copy_gt_disc_labels(){
+  local file="$1"
+  local type="$2"
+  local contrast="$3"
+
+  if [[ $contrast == "T1w" ]] || [[ $contrast == "T2w" ]]; then
+    file_name="${file%%_*}_${contrast}_label-discs_dlabel"
+  elif [[ $contrast == "T2star" ]]; then
+    file_name="${file%%_*}_${contrast}_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "MTon" ]]; then
+    file_name="${file%%_*}_flip-1_mt-on_MTS_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "MToff" ]]; then
+    file_name="${file%%_*}_flip-2_mt-off_MTS_label-discs_desc-warp_dlabel"
+  elif [[ $contrast == "DWI" ]]; then
+    file_name="${file%%_*}_rec-average_dwi_label-discs_desc-warp_dlabel"
+  fi
+  # Construct file name to GT segmentation located under derivatives/labels
+  FILEDISCLABELS="${PATH_DATA}/derivatives/labels/${SUBJECT}/${type}/${file_name}.nii.gz"
+  echo ""
+  echo "Looking for manual disc labels: $FILEDISCLABELS"
+  if [[ -e $FILEDISCLABELS ]]; then
+      echo "Found! Copying ..."
+      rsync -avzh $FILEDISCLABELS ${file}_discs.nii.gz
+  else
+      echo "File ${FILEDISCLABELS} does not exist" >> ${PATH_LOG}/missing_files.log
+      echo "ERROR: Manual Disc Labels ${FILEDISCLABELS} does not exist. Exiting."
+      exit 1
+  fi
+}
+
+
+# Copy GT soft binarized segmentation (located under derivatives/labels_softseg_bin)
+copy_gt_softseg_bin(){
+  local file="$1"
+  local type="$2"
+  # Construct file name to GT segmentation located under derivatives/labels_softseg_bin
+  # NOTE: the naming conventions are in the revised BIDS format
+  FILESEG="${PATH_DATA}/derivatives/labels_softseg_bin/${SUBJECT}/${type}/${file}_desc-softseg_label-SC_seg.nii.gz"
+  echo ""
+  echo "Looking for manual segmentation: $FILESEG"
+  if [[ -e $FILESEG ]]; then
+      echo "Found! Copying ..."
+      rsync -avzh $FILESEG ${file}_softseg_bin.nii.gz
+  else
+      echo "File ${FILESEG} does not exist" >> ${PATH_LOG}/missing_files.log
+      echo "ERROR: Manual Segmentation ${FILESEG} does not exist. Exiting."
+      exit 1
+  fi
+}
+
+
+# Segment spinal cord
+segment_sc(){
+  local file="$1"
+  local file_gt_vert_label="$2"
+  local model_basename="$3"     # 2d or 3d
+  local contrast="$4"   # used only for saving output file name
+  # local kernel="$5"    # 2d or 3d_fullres
+
+  FILESEG="${file%%_*}_${contrast}_seg_${model_basename}"
+
+  # Get the start time
+  start_time=$(date +%s)
+  # # Run SC segmentation
+  # python ${PATH_NNUNET_SCRIPT} -i ${file}.nii.gz -o ${FILESEG}.nii.gz -path-model ${PATH_NNUNET_MODEL}/nnUNetTrainer__nnUNetPlans__${kernel} -pred-type sc -use-gpu -use-best-checkpoint
+  # Run SC segmentation (natively with sct_deepseg)
+  sct_deepseg spinalcord -i ${file}.nii.gz -o ${FILESEG}.nii.gz -qc ${PATH_QC} -qc-subject ${SUBJECT}
```

Comment:
should we remove the first line with python, just to clean up

Reply:
done here! https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/139/commits/23151194d2f7810f4112be7cc5ee886fb9310516